modules/javafx.controls/src/main/java/javafx/scene/control/skin/ToolBarSkin.java

```diff
@@ -552,17 +560,35 @@ public CssMetaData<ToolBar,Pos> getCssMetaData() {
     private void initialize() {
         if (getSkinnable().getOrientation() == Orientation.VERTICAL) {
             box = new VBox();
+            overflowBox = new VBox();
         } else {
             box = new HBox();
+            overflowBox = new HBox();
         }
         box.getStyleClass().add("container");
         box.getChildren().addAll(getSkinnable().getItems());
-        overflowMenu = new ToolBarOverflowMenu(overflowMenuItems);
+        // The overflowBox must have the same style classes, otherwise the overflow items may get wrong values.
+        overflowBox.idProperty().bind(box.idProperty());
+        overflowBox.getStyleClass().setAll(box.getStyleClass());
+        box.getStyleClass().addListener((ListChangeListener<? super String>) change -> overflowBox.getStyleClass().setAll(change.getList()));
+        overflowBox.getStylesheets().setAll(box.getStylesheets());
+        box.getStylesheets().addListener((ListChangeListener<? super String>) change -> overflowBox.getStylesheets().setAll(change.getList()));
```

Comment:
this is interesting.  

isn't `overflowBox` a sibling of `box`, having the same parent, and therefore inheriting the same set of stylesheets from the parent `Scene`?

Reply:
In addition to those from the scene, each ``Parent`` can have its own stylesheets.

Reply:
OK, you are correct - the `box` is technically visible via node lookup by the "content" style, so the app developer can attach a stylesheet to it (why they would do it is another question, I am sure there are many other places where doing so would break the code).
And we are not creating leaks because both nodes are siblings.